qt/shutdowndlg.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-FileCopyrightText: © 2025 Huaide Jiang
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# This file is part of the program "Back In Time" which is released under GNU
+# General Public License v2 (GPLv2). See LICENSES directory or go to
+# <https://spdx.org/licenses/GPL-2.0-or-later.html>.
+""" Module about showing a warning dialog before shutting down """
+import gettext
+from PyQt6.QtWidgets import (QDialog,
+                             QLabel,
+                             QPushButton,
+                             QVBoxLayout,
+                             QHBoxLayout)
+from PyQt6.QtCore import QTimer, Qt
+
+
+class ShutdownWarningDlg(QDialog):
+    """A UI class for a shutting down window."""
+    def __init__(self, countdown):
+        super().__init__()
+        self.countdown = countdown
+
+        # Initialize UI components
+        self.setWindowTitle(_('Countdown to Shutdown'))
+        self.label = QLabel(gettext.ngettext(
+            'Finished backup, shutdown in {n} second.',
+            'Finished backup, Shutdown in {n} seconds.',
+            self.countdown).format(n=self.countdown), self)
+        # Center the label text
+        self.label.setAlignment(Qt.AlignmentFlag.AlignCenter)
+        self.cancel_button = QPushButton(_('Cancel Shutdown'), self)
+        self.shutdown_button = QPushButton(_('Shutdown Now'), self)
+        self.cancel_button.clicked.connect(self.cancel_shutdown)
+        # Immediately accept on shutdown now
+        self.shutdown_button.clicked.connect(self.accept)
+
+        # Layout setup
+        layout = QVBoxLayout()
+        # Add stretch before label to center it vertically
+        layout.addStretch(1)
+        layout.addWidget(self.label)
+        # Add stretch after label to center it vertically
+        layout.addStretch(1)
```

Comment:
Try without stretch elements. For example:
```suggestion
        layout.addWidget(self.label, alignment=Qt.AlignmentFlag.AlignHCenter)
```